theano/gpuarray/blas.py

```diff
@@ -244,8 +256,11 @@ def make_node(self, A, alpha, x, y):
         A = as_gpuarray_variable(A, ctx_name)
         x = as_gpuarray_variable(x, ctx_name)
         y = as_gpuarray_variable(y, ctx_name)
-        with theano.configparser.change_flags(warn_float64='ignore'):
-            alpha = as_tensor_variable(alpha).astype('float64')
+        alpha = as_tensor_variable(alpha)
+        if len(set([A.dtype, alpha.dtype, x.dtype, y.dtype])) != 1:
```

Comment:
why? This seem much clear to my point of view? This is copied code from tensor/blas.py anyway...